projects/wp-nextjs/tsconfig.json

```diff
@@ -13,8 +12,8 @@
     "noEmit": true,
     "incremental": true,
     "esModuleInterop": true,
-    "module": "esnext",
-    "moduleResolution": "node16",
+    "module": "ESNext",
+    "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
     "jsx": "preserve"
```

Comment:
A couple additions that may be helpful, but not necessarily add too much strictness to the code. More around good coding practices rather than TS as such.
- `noFallthroughCasesInSwitch: true`- [Definition](https://www.typescriptlang.org/tsconfig/#noFallthroughCasesInSwitch) - ensures you are not skipping anything in a switch statement, and always at least provide a `default` option.
- `noImplicitAny` - [Definition](https://www.typescriptlang.org/tsconfig/#noImplicitAny) - ensures you define as `any` explicitly, rather than implicitly assume it. This may add a bunch more changes to this PR however, but worth checking.
- `strictBindCallApply` - [Definition](https://www.typescriptlang.org/tsconfig/#strictBindCallApply) - can be helpful.
- `strictNullChecks` - [Definition](https://www.typescriptlang.org/tsconfig/#strictNullChecks) - as the name implies, stricter null checks.
- `noUnusedParameters` - [Definition](https://www.typescriptlang.org/tsconfig/#noUnusedParameters) - also as the name implies, does not allow unused parameters to remain in code.

Reply:
I think it would be nice to have a tsconfig.strict.json and tsconfig.json. The code in this repo would pass the strict tsconfig. Projects can choose whether they want to enforce strict TS or not depending on the engineers on the project.

Reply:
`noFallthroughCasesInSwitch` and `noUnusedParameters` have very similar eslint rules in the 10up eslint config so I see no harm in adding it to TS. `strictBindCallApply` I think it's helpful and we don't write code using bind, .call very often so don't think it will make things too strict most of the time and `strictNullChecks` can help catch bugs so I think it makes sense.

`noImplicitAny` is the most strick one IMO but based on our convo we do want `any` to be a escape hatch for unexperience engineers so I'm fine forcing them to explicitly set `any` when they do need the escape hatch which would also make it easier to come back and update types later.

I don't think two configs would make sense as it would just add more confusion IMO.